### ISSUE TITLE
Sensor reinitialization memory leak

### ISSUE THREAD

**Issue by adamgarbo** [adafruit/Adafruit_LIS2MDL#11]
Hey folks,

It appears that the LIS2MDL library is subject to the same type of memory leak I found in the LSM6DS library: https://github.com/adafruit/Adafruit_LSM6DS/issues/28

Similar to my last issue, this issue wasn't discovered until my instruments began to reset themselves after 766 reinitializations. As the leak has a size of 32 bytes, it takes a bit longer to crash the stack when working with the SAMD21.

I can also confirm this issue is not present in the `<Adafruit_LSM303_Accel.h>` library.

Cheers,
Adam

Arduino board: [Adafruit LSM303AGR Accelerometer Magnetometer](https://www.adafruit.com/product/4413)

Arduino IDE version: **1.18.19**

Steps to reproduce the problem: **Please see below.**

**Example Code:**

```
#include <Wire.h>
#include <Adafruit_Sensor.h>
#include <Adafruit_LIS2MDL.h>

Adafruit_LIS2MDL lis2mdl = Adafruit_LIS2MDL(12345);

void setup() {

  Serial.begin(115200);
  while (!Serial);

  pinMode(A4, OUTPUT);
  digitalWrite(A4, LOW);
}

void loop() {
  Wire.begin(); // Initialize I2C

  digitalWrite(A4, HIGH); // Enable power
  delay(500);

  configureMag();

  digitalWrite(A4, LOW); // Disable power
  delay(500);

  Serial.print(" freeRam(): "); Serial.println(freeRam());
  delay(5000);
}

void configureMag() {

  if (!lis2mdl.begin()) {
    Serial.println("Failed to find LIS2MDL chip");
  }
  else {
    Serial.print("Initialized LIS2MDL");
  }
}

// Function to print available 32K SRAM memory
extern "C" char *sbrk(int i);
int freeRam()
{
  char stack_dummy = 0;
  return &stack_dummy - sbrk(0);
}
```

**Feather M0 Proto + LSM303AGR**

Pin powered by A4.
**Output:**

```
15:05:25.015 -> Initialized LIS2MDL freeRam(): 27927
15:05:31.232 -> Initialized LIS2MDL freeRam(): 27895
15:05:37.440 -> Initialized LIS2MDL freeRam(): 27863
15:05:43.631 -> Initialized LIS2MDL freeRam(): 27831
15:05:49.813 -> Initialized LIS2MDL freeRam(): 27799
15:05:56.013 -> Initialized LIS2MDL freeRam(): 27767
15:06:02.214 -> Initialized LIS2MDL freeRam(): 27735
15:06:08.442 -> Initialized LIS2MDL freeRam(): 27703
15:06:14.629 -> Initialized LIS2MDL freeRam(): 27671
15:06:20.840 -> Initialized LIS2MDL freeRam(): 27639
15:06:27.036 -> Initialized LIS2MDL freeRam(): 27607
15:06:44.244 -> Initialized LIS2MDL freeRam(): 27575
15:06:44.244 -> Initialized LIS2MDL freeRam(): 27543
15:06:45.624 -> Initialized LIS2MDL freeRam(): 27511
15:06:51.818 -> Initialized LIS2MDL freeRam(): 27479
15:06:58.042 -> Initialized LIS2MDL freeRam(): 27447
15:07:04.224 -> Initialized LIS2MDL freeRam(): 27415
15:07:10.443 -> Initialized LIS2MDL freeRam(): 27383
15:07:17.277 -> Initialized LIS2MDL freeRam(): 27351
```

**Comment by adamgarbo** [comment]
Looks like the issue is with the creation of `config_a`:

https://github.com/adafruit/Adafruit_LIS2MDL/blob/819cf7483f1c0f8f982bc1c2e6ef2f260b3c0579/Adafruit_LIS2MDL.cpp#L176-L177

Adding a simple `if (config_a) delete config_a;` before line 176 seems to do the trick.

```
16:28:44.885 -> Initialized LIS2MDL freeRam(): 27959
16:28:51.092 -> Initialized LIS2MDL freeRam(): 27959
16:28:57.291 -> Initialized LIS2MDL freeRam(): 27959
16:29:03.502 -> Initialized LIS2MDL freeRam(): 27959
16:29:09.768 -> Initialized LIS2MDL freeRam(): 27959
16:29:15.917 -> Initialized LIS2MDL freeRam(): 27959
16:29:22.086 -> Initialized LIS2MDL freeRam(): 27959
16:29:28.324 -> Initialized LIS2MDL freeRam(): 27959
16:29:34.495 -> Initialized LIS2MDL freeRam(): 27959
16:29:40.778 -> Initialized LIS2MDL freeRam(): 27959
16:29:46.930 -> Initialized LIS2MDL freeRam(): 27959
```
Cheers,
Adam

**Comment by adamgarbo** [comment]
Addressed by #12.
